mmaction/core/bbox/transforms.py

```diff
@@ -3,35 +3,58 @@
 
 
 def bbox2result(bboxes, labels, num_classes, thr=0.01):
-    """Convert detection results to a list of numpy arrays.
+    """
+    Convert detection results to a list of numpy arrays.
+
+    This identifies single-label classification (as opposed to multi-label)
+    through the thr parameter which is set to a negative value.
+
+    Currently, the way to set this is to set
+       `test_cfg.rcnn.action_thr=-1.0`
+    The ideal way, however, which is not yet implemented, is to have another
+    metric, mAP@top1 and set that within:
+       `evaluation.metrics=[...,'mAP@top1',...]`
 
     Args:
         bboxes (Tensor): shape (n, 4)
         labels (Tensor): shape (n, #num_classes)
         num_classes (int): class number, including background class
         thr (float): The score threshold used when converting predictions to
-            detection results
+            detection results. If a single negative value, uses single-label
+            classification
     Returns:
         list(ndarray): bbox results of each class
     """
     if bboxes.shape[0] == 0:
         return list(np.zeros((num_classes - 1, 0, 5), dtype=np.float32))
 
     bboxes = bboxes.cpu().numpy()
-    labels = labels.cpu().numpy()
+    scores = labels.cpu().numpy()  # rename for clarification
+
+    # Although we can handle single-label classification, we still want scores
+    assert scores.shape[-1] > 1
 
-    # We only handle multilabel now
-    assert labels.shape[-1] > 1
+    # Robustly check for multi/single-label:
+    if not hasattr(thr, '__len__'):
+        multilabel = thr >= 0
+        thr = (thr,) * num_classes
+    else:
+        multilabel = False
```

Comment:
I think it should be `multilabel = True` here. Since the default setting for the original code is `multilabel = True`.

Reply:
Actually, you are right: that was a bug in my logic. Rather, it should be True, because when we have multiple individual thresholds it explicitly should indicate that we are multi-label. Thanks. Fixed in the next commit.